src/cli/serve/read_yaml_config.js

```diff
@@ -1,7 +1,16 @@
-import { isArray, isPlainObject, forOwn, set, transform } from 'lodash';
+import { isArray, isPlainObject, forOwn, set, transform, isString } from 'lodash';
 import { readFileSync as read } from 'fs';
 import { safeLoad } from 'js-yaml';
 
+function _replaceEnvVarRefs(val) {
+  return val.replace(/\$\{(\w+)\}/g, (match, envVarName) => {
+    if (process.env[envVarName] !== undefined) {
+      return process.env[envVarName];
```

Comment:
I would prefer if we _.escaped this by default, and any configuration that needs it unescaped can do so.  Thoughts?

Reply:
@jbudz 
So you want to replace `return process.env[envVarName];` 
by `return _.escape(process.env[envVarName]);` ?

No problem for me, but can you explain me why ? I mean : at this point, the yaml file is already parsed.

Maybe it could be an issue for `server.basePath` option if user wants to use `&` character ?

Reply:
Yeah, I don't think we want to escape HTML entities in the config values. People using this feature need to take care that they inject the correct value and Joi will take care of ensuring that it's a valid value.

I was initially concerned about non-string values, but realized Joi also parses JSON values in an attempt to coerce the value to the desired type, so to set a boolean value you can do:

```yaml
some.boolean: "${MY_ENV_VAR}"
```

and
 
```sh
MY_ENV_VAR=true bin/kibana
```

And Joi will correctly convert the valid JSON so that

```js
config.get('some.boolean') === true
```

Reply:
in practice I can't think of anything realistic, it's mostly how I operate.  I'l  defer on this.

Reply:
I agree with @spalger 
In an option like `elasticsearch.customHeaders`, people want that the final value is exactly what they type. And value can contain some special characters like `&`.